### PR TITLE
Spawning vehicles

### PR DESCRIPTION
**Pull request by darknight2590**
In writing these posts, I relied on Spawning A Vehicle At A Distance in discord server
The following notes through the words of Mr.Kakarot

This will spawn the entity and callback the entity id which you can then use on the client side to check if the entity exists. It will NOT exist until a client(player) gets within range of it! While a client is NOT in range, the server “owns” the entity. So whenever a client gets into infinity range it then takes ownership of that entity and can be used by them.

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? [yes] (Be honest)
- Does your code fit the style guidelines? [yes]
- Does your PR fit the contribution guidelines? [yes]

native using :
https://docs.fivem.net/natives/?_0x6E31E993
https://docs.fivem.net/natives/?_0x3AC90869
https://docs.fivem.net/natives/?_0x98EFF6F1
https://docs.fivem.net/natives/?_0x1647F1CB
https://docs.fivem.net/natives/?_0x972CC383

### CODE REVIEW

File: server/functions.lua [hunk 1]
```diff
@@ -209,6 +209,21 @@ function PaycheckLoop()
     SetTimeout(QBCore.Config.Money.PayCheckTimeOut * (60 * 1000), PaycheckLoop)
 end
 
+QBCore.Functions.SpawnVehicle = function(source, entity, locations, plate)
+    local CreateAutomobile = `CREATE_AUTOMOBILE`
+    local model = (type(entity) == "number" and entity or GetHashKey(entity))
```
Comment (Mojito-Fivem) [inline server/functions.lua:214]: this is not a valid way of getting model hash from an entity, and if this is a spawn vehicle function - why are you using the entity handle as a parameter??

File: server/functions.lua [hunk 2]
```diff
@@ -209,6 +209,21 @@ function PaycheckLoop()
...
+    local coords = locations ~= nil and locations or GetEntityCoords(GetPlayerPed(source))
+    local heading = locations ~= nil and locations.w ~= nil and locations.w or GetEntityHeading(GetPlayerPed(source))
+    local vehicle = Citizen.InvokeNative(CreateAutomobile, model, coords.x, coords.y, coords.z, heading);
+    while not DoesEntityExist(vehicle) do
+        Wait(0)
```
Comment (Mojito-Fivem) [inline server/functions.lua:219]: this can be improved with promises

File: server/functions.lua [hunk 3]
```diff
@@ -209,6 +209,21 @@ function PaycheckLoop()
     SetTimeout(QBCore.Config.Money.PayCheckTimeOut * (60 * 1000), PaycheckLoop)
 end
 
+QBCore.Functions.SpawnVehicle = function(source, entity, locations, plate)
+    local CreateAutomobile = `CREATE_AUTOMOBILE`
+    local model = (type(entity) == "number" and entity or GetHashKey(entity))
+    local coords = locations ~= nil and locations or GetEntityCoords(GetPlayerPed(source))
+    local heading = locations ~= nil and locations.w ~= nil and locations.w or GetEntityHeading(GetPlayerPed(source))
```
Comment (Mojito-Fivem) [inline server/functions.lua:216]: this `locations` variable could do with a better name

File: server/functions.lua [hunk 4]
```diff
@@ -209,6 +209,21 @@ function PaycheckLoop()
...
+    local vehicle = Citizen.InvokeNative(CreateAutomobile, model, coords.x, coords.y, coords.z, heading);
+    while not DoesEntityExist(vehicle) do
+        Wait(0)
+    end
+    if plate ~= nil and plate ~= false then
+        SetVehicleNumberPlateText(vehicle, plate)
+        return vehicle
```
Comment (Mojito-Fivem) [inline server/functions.lua:223]: It would make more sense if you returned NETWORK ID and not Entity Handle

**Review by Mojito-Fivem — CHANGES_REQUESTED** [review]
Please, for the love of cheese, comment your code.

**Comment by darknight2590** [comment]
I wrote it with my own little experience I think you have a better idea that you can prepare better
I am writing it according to my ability to do it and everyone can participate in making it better

**Comment by Mojito-Fivem** [comment]
I've requested changes to this PR, this isn't about me having better ideas - this is about maintaining qb-core to a high standard.

**Comment by IdrisDose** [comment]
Process is being changed, thank you for this PR though.

**Pull request closed without merging**